### PR TITLE
Improve  user type

### PR DESCRIPTION
**Pull request by tom-ee**
3 improvements for __user:

* redirect stdout and error of `userdel` to get rid of annoying output
* handle typos in `--state`-parameter gracefully (previously `--state presetn` or similar typos would have delete the user
* message user removal (w/ and w/o remove-home)

manpage updated accordingly.

### CODE REVIEW

File: cdist/conf/type/__user/gencode-remote [hunk 1]
```diff
@@ -130,13 +131,17 @@ if [ "$state" = "present" ]; then
           echo useradd "$@" "$name"
        fi
     fi
-else
+elif [ "$state" = "absent" ]; then
     if grep -q "^${name}:" "$__object/explorer/passwd"; then
         #user exists, but state != present, so delete it
         if [ -f "$__object/parameter/remove-home" ]; then
-            echo userdel -r "${name}"
+            printf "userdel -r %s >/dev/null 2>&1\n" "${name}"
```
Comment (thriqon) [inline cdist/conf/type/__user/gencode-remote]: I'd suggest single quotes here, so the name arrives quoted on the target.

File: cdist/conf/type/__user/gencode-remote [hunk 2]
```diff
@@ -130,13 +131,17 @@ if [ "$state" = "present" ]; then
...
+	    echo "userdel -r" >> "$__messages_out"
         else
-            echo userdel "${name}"
+            printf "userdel %s >/dev/null 2>&1\n" "${name}"
```
Comment (thriqon) [inline cdist/conf/type/__user/gencode-remote]: see above, single quotes?

File: cdist/conf/type/__user/gencode-remote (hunk 1 above)
Comment (tom-ee) [reply cdist/conf/type/__user/gencode-remote]: I was hesitant about this one, as users can (should?) not contain whitespace -- but better safe that sorry here ...

**Comment by darko-poljak** [comment]
@asteven @telmich Any remarks/suggestions?

**Pull request merged**